src/libraries/System.Private.CoreLib/src/System/Globalization/OrdinalCasing.Icu.cs

```diff
@@ -85,7 +92,12 @@ internal static char ToUpper(char c)
                 return (char) s_basicLatin[(int)c];
             }
 
-            ushort[]? casingTable = s_casingTable[pageNumber];
+            if (s_casingTable == null)
+            {
+                InitCasingTable();
+            }
+
+            ushort[]? casingTable = s_casingTable![pageNumber];
```

Comment:
instead of adding `!` you can add `[MemberNotNull(nameof(s_casingTable))]` attribute to `InitCasingTable`